core/src/modules/middleware/authentication.ts

```diff
@@ -218,3 +220,17 @@ async function authenticateTeamMemberInRoom(
     }
   });
 }
+
+// check if the apiKey was sent via the header
+function parseHeaders(data: ActionProcessor<any>) {
+  if (data.connection.type !== "web") return;
+
+  if (data.connection.rawConnection.req.headers["x-grouparoo-api-key"]) {
+    data.params.apiKey =
+      data.connection.rawConnection.req.headers["x-grouparoo-api-key"];
```

Comment:
My only concern about the implementation here is that you're the data object's parsed `?apiKey=` to assign the value the header, therefore you'd have to understand the implementation of this function to know where the value is really coming from.

The ideal case would be to have the apiKey come from either params or the header itself, but that would take a lot more code changes. Same for the server token below.

Reply:
Well, the goal was to allow for both - the user can choose to send the key and server token either via header or query param